src/proj_4D_api.c

```diff
@@ -724,22 +724,74 @@ PJ_CONTEXT *proj_context_destroy (PJ_CONTEXT *ctx) {
 }
 
 
+
+
+
+
 /*****************************************************************************/
-PJ_INFO proj_info(void) {
+static char *path_append (char *buf, const char *app, size_t *buf_size) {
+/******************************************************************************
+    Helper for proj_info() below. Append app to buf, separated by a
+    semicolon. Also handle allocation of longer buffer if needed.
+
+    Returns buffer and adjusts *buf_size through provided pointer arg.
+******************************************************************************/
+    char *p;
+    size_t len, applen = 0, buflen = 0;
+
+    /* Nothing to do? */
+    if (0 == app)
+        return buf;
+    applen = strlen (app);
+    if (0 == applen)
+        return buf;
+
+    /* Start checking whether buf is long enough */
+    if (0 != buf)
+        buflen = strlen (buf);
+    len = buflen+applen+2;
+
+    /* "pj_realloc", so to speak */
+    if (*buf_size < len) {
+        p = pj_calloc (2 * len, sizeof (char));
+        if (0==p) {
+            pj_dealloc (buf);
+            return 0;
+        }
+        *buf_size = 2 * len;
+        if (buf != 0)
+            strcpy (p, buf);
+        buf = p;
+    }
+
+    /* Only append a semicolon if something's already there */
+    if (0 != buflen)
+        strcat (buf, ";");
+    strcat (buf, app);
+    return buf;
+}
+
+static const char *empty = {""};
+static char version[64]  = {""};
+static PJ_INFO info = {0, 0, 0, 0, 0, 0};
+
+
+/*****************************************************************************/
+PJ_INFO proj_info (void) {
 /******************************************************************************
     Basic info about the current instance of the PROJ.4 library.
 
     Returns PJ_INFO struct. Searchpath member of the struct is truncated to 512
     characters.
-
 ******************************************************************************/
-    PJ_INFO info;
     const char * const *paths;
-    char *tmpstr;
     int i, n;
-    size_t len = 0;
 
-    memset(&info, 0, sizeof(PJ_INFO));
+    size_t  buf_size = 0;
+    char   *buf = 0;
+
+    if (0 != info.version)
```

Comment:
Be careful: this is not thread-safe. Ideally this should be protected by a mutex

Reply:
Wow - that was really a blunder. Thanks for spotting it!